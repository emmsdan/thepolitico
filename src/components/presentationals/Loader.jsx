import React from 'react';
import PropTypes from 'prop-types';

import logo from '../../assets/images/official/favicon.png';

const styleLoader = {
  backgroundImage: `url(${logo})`,
  backgroundSize: 'contain',
  backgroundRepeat: 'no-repeat',
  borderRadius: 0,
};

const Loader = props => {
  return (
    <div className={`center load-overlay ${props.loader}`}>
      <div>
        <div className="loader center" style={styleLoader} />
        please waiting am loading content...
      </div>
    </div>
  );
};

Loader.propTypes = {
  loader: PropTypes.string,
};

export default Loader;
