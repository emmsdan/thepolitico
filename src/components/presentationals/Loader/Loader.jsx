import React from 'react';
import PropTypes from 'prop-types';

import logo from '../../../assets/images/official/favicon.png';

const styleLoader = {
  backgroundImage: `url(${logo})`,
  backgroundSize: 'contain',
  backgroundRepeat: 'no-repeat',
  borderRadius: 0,
};

const Loader = ({ loader }) => {
  return (
    <div className={`center load-overlay ${loader}`}>
      <div>
        <div className="loader center" style={styleLoader} />
        <strong>please waiting am loading content...</strong>
      </div>
    </div>
  );
};

Loader.propTypes = {
  loader: PropTypes.string,
};

export default Loader;
