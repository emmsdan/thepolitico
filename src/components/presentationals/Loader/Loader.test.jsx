import React from 'react';
import { shallow } from 'enzyme';

import Loader from './Loader';

describe('<Loader />', () => {
  it('should render correctly', () => {
    const component = shallow(<Loader loader="show" />);
    expect(component).toMatchSnapshot();
  });
});
