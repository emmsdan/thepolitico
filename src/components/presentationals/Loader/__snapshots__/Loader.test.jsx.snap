// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render correctly 1`] = `
<div
  className="center load-overlay show"
>
  <div>
    <div
      className="loader center"
      style={
        Object {
          "backgroundImage": "url([object Object])",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "contain",
          "borderRadius": 0,
        }
      }
    />
    <strong>
<<<<<<< HEAD
      please waiting am loading content
      ...
=======
      please waiting am loading content...
>>>>>>> tested loader component
    </strong>
  </div>
</div>
`;
